pkg/kv/kvserver/store_split.go

```diff
@@ -350,16 +350,20 @@ func (s *Store) SplitRange(
 		// assumption that distribution across all tracked load stats is
 		// identical.
 		leftRepl.loadStats.Split(rightRepl.loadStats)
-		if err := s.addToReplicasByKeyLocked(rightRepl); err != nil {
+		rightRepl.mu.RLock()
```

Comment:
Why not `defer rightRepl.mu.RUnlock()`? If anyone adds a new `return err` below chances are they're going to miss it.

But also, can you unlock before `updateRangeInfo`? That method doesn't need the lock so someone might change it and it could then call something like `.Desc()` and cause deadlocks.

You can hold on to the descriptor returned from `descRLocked()` even after releasing the mutex. (This might be worth pointing out on `Desc()` and `descRLocked()`).

Reply:
`updateRangeInfo` does call `r.mu.Lock()`, and caused a deadlock, so I unlock now. Using `rightDesc` instead of `rightRepl.Desc()` too, it's the one used to init this replica.

Did not `defer` the unlock yet, looking at this.

Reply:
Leaving the lock/unlock as is, now its scope is obvious (one `addToReplicas...` call). I'm revisiting/shuffling this code further anyway, so this might change.